modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-order-impl/src/main/java/com/liferay/headless/commerce/admin/order/internal/resource/v1_0/OrderResourceImpl.java

```diff
@@ -103,6 +104,11 @@ public class OrderResourceImpl extends BaseOrderResourceImpl {
 
 	@Override
 	public Response deleteOrder(Long id) throws Exception {
+		ServiceContext serviceContext =
+			ServiceContextThreadLocal.getServiceContext();
```

Comment:
hey @dsitu why do we need this?

Reply:
Hey @riccardo-alberti im not sure if this would be the best way to fix this, but I found a bug yesterday while running the specific functional test that was failing on my local instance.
This test: **CommerceMiniCart#CanUseMiniCartWidgetToCheckoutWithBuyerOrderApprovalWorkflow**

When "Buyer Order Approval Workflow" is set to "Single Approver(Version 1), and a new order is created and work-flowed to the end, when I try to delete that order with the api call "http://localhost:8080/o/headless-commerce-admin-order/v1.0/orders/489631", it would throw an error:
```
{
  "status": "NOT_FOUND",
  "title": "No User exists with the primary key 0"
}
```
Inside CommerceOrderItemLocalService.deleteCommerceOrderItem(id) it tries to update the workflow of the parent CommerceOrder and it tries to pull the UserId from serviceContext. When we try to call it from the rest endpoint the userId is not set.

This can only be tested when trying to call the api manually. It will work just fine if you try testing on "Liferay API Explorer".

Reply:
Hey @dsitu there's already an issue created for this problem https://issues.liferay.com/browse/COMMERCE-10011

Reply:
@victoroscuro oh nice, I will remove that commit from this PR and rerun ci:test. Thanks victor!